### DISCUSSION TITLE
Problems compiling in Ubuntu 20.04 LTS

### DISCUSSION THREAD

**Discussion by nonogrub** [madMAx43v3r/chia-plotter#146] [Q&A]
[ 99%] Building CXX object CMakeFiles/chia_plot.dir/src/chia_plot.cpp.o
In file included from /home/chia/chia-plotter/lib/bls-signatures/src/elements.hpp:22,
                 from /home/chia/chia-plotter/lib/bls-signatures/src/privatekey.hpp:24,
                 from /home/chia/chia-plotter/lib/bls-signatures/src/bls.hpp:18,
                 from /home/chia/chia-plotter/src/chia_plot.cpp:15:
/home/chia/chia-plotter/lib/bls-signatures/src/util.hpp: In static member function ‘static std::string bls::Util::HexStr(const uint8_t*, size_t)’:
/home/chia/chia-plotter/lib/bls-signatures/src/util.hpp:62:25: warning: comparison of integer expressions of different signedness: ‘int’ and ‘size_t’ {aka ‘long unsigned int’} [-Wsign-compare]
   62 |         for (int i=0; i < len; ++i)
      |                       ~~^~~~~
/home/chia/chia-plotter/lib/bls-signatures/src/util.hpp: In static member function ‘static std::string bls::Util::HexStr(const std::vector<unsigned char>&)’:
/home/chia/chia-plotter/lib/bls-signatures/src/util.hpp:70:25: warning: comparison of integer expressions of different signedness: ‘int’ and ‘std::vector<unsigned char>::size_type’ {aka ‘long unsigned int’} [-Wsign-compare]
   70 |         for (int i=0; i < data.size(); ++i)
      |                       ~~^~~~~~~~~~~~~
[100%] Linking CXX executable test_phase_3
[100%] Built target test_phase_3
[100%] Linking CXX executable test_phase_1
[100%] Built target test_phase_1
[100%] Linking CXX executable chia_plot
[100%] Built target chia_plot

**Comment by j4r3kb** [comment]
Ubuntu 18.04 with compiled cmake version 3.20.20210610-g5fd68d3
```
/usr/bin/ar: creating /home/plotter/build/libbls.a
[ 98%] Built target combined_custom
[ 99%] Building CXX object lib/bls-signatures/src/CMakeFiles/bls.dir/privatekey.cpp.o
[ 99%] Linking CXX static library libbls.a
[ 99%] Built target bls
[ 99%] Building CXX object CMakeFiles/chia_plot.dir/src/chia_plot.cpp.o
In file included from /home/plotter/lib/bls-signatures/src/elements.hpp:22:0,
                 from /home/plotter/lib/bls-signatures/src/privatekey.hpp:24,
                 from /home/plotter/lib/bls-signatures/src/bls.hpp:18,
                 from /home/plotter/src/chia_plot.cpp:15:
/home/plotter/lib/bls-signatures/src/util.hpp: In static member function ‘static std::__cxx11::string bls::Util::HexStr(const uint8_t*, size_t)’:
/home/plotter/lib/bls-signatures/src/util.hpp:62:25: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
         for (int i=0; i < len; ++i)
                       ~~^~~~~
/home/plotter/lib/bls-signatures/src/util.hpp: In static member function ‘static std::__cxx11::string bls::Util::HexStr(const std::vector<unsigned char>&)’:
/home/plotter/lib/bls-signatures/src/util.hpp:70:25: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
         for (int i=0; i < data.size(); ++i)
                       ~~^~~~~~~~~~~~~
[100%] Linking CXX executable test_phase_3
[100%] Built target test_phase_3
[100%] Linking CXX executable test_phase_1
[100%] Built target test_phase_1
[100%] Linking CXX executable chia_plot
[100%] Built target chia_plot

```

**Reply by madMAx43v3r** [reply]
those are warnings, nothing to worry about

**Comment by nonogrub** [comment]
I tried to run with compile errors but the times are.....   i7 10700 32GB RAM DDR4

Multi-threaded pipelined Chia k32 plotter - 18f4918
Final Directory: /mnt/hdd/
Number of Plots: 1
Process ID: 3800
Number of Threads: 16
Number of Buckets: 2^7 (128)
Pool Public Key:   xxxx
Farmer Public Key: yyyyyy
Working Directory:   /mnt/ssd/
Working Directory 2: /mnt/ssd/
Plot Name: plot-k32-2021-06-10-15-xxxxxxxxxxxxxxx
[P1] Table 1 took 79.7945 sec

**Comment by nonogrub** [comment]
I tried to install via PIP the latest version of BLAKE3 but when I launch Mad Max, at the end of the P1 table occur this:
[P1] Table 1 took 500.497 sec
terminate called after throwing an instance of 'std::runtime_error'
  what():  thread failed with: thread failed with: thread failed with: thread failed with: fwrite() failed
Aborted

My Ubuntu version is 20.04 LTS fresh install

**Reply by madMAx43v3r** [reply]
that usually means lack of disk space
